### ISSUE TITLE
Represent the gateway itself as a web thing

### ISSUE THREAD

**Issue by benfrancis** [mozilla-iot/gateway#2146]
As a developer I want the gateway itself to be represented as a web thing so I can perform actions on the gateway and enumerate the web things it manages via the Web Thing API

See also: https://github.com/mozilla-iot/schemas/issues/35

**Comment by flatsiedatsie** [comment]
This would be of interest to me for two reaons:
- Voco currently tries to detect gateways on the network. It would be great if it could detect which gateways are also running Voco, and whether these are in a "controller" or "satellite" state.
- If gateways could share their data, it would be possible to segment the home network. Children could then have their own gateway, which might help with allowing them some level of additional control over devices in their rooms. (Although user accounts with additional controls would also be good route in their direction, and require less extra hardware).
- I am currently working on an addon which also creates a webthings on port 8888 because I needed the thing to be accessible to other gateways on the network. I realise this is sub-optimal, as it would mean that port 8888 could already be taken (assuming port 8888 is the designated webthings port, which I'm not sure about). If the gateway could (selectively) make its devices available to the network, then this extra webserver could be removed in favour of the gateway's own centralised functionality.

In other words: I'm all for this :-)
